### ISSUE TITLE
Problem with FFTGraph in fft.py - set_color with a list instead of a single string

### ISSUE THREAD

**Issue by mrmcsoftware** [nipunramk/Reducible#2]
I'm having a problem running your FFT/fft.py (FFTGraph scene).  set_color is expecting a string rather than a list of strings and is therefore producing the following error:

       edges[i].set_color(color=[BLUE, SKY_BLUE])
    File "/home/mark/.local/lib/python3.8/site-packages/manim/mobject/types/vectorized_mobject.py", line 283, in set_color
       self.color = colour.Color(color)
    File "/home/mark/.local/lib/python3.8/site-packages/colour.py", line 988, in __init__
       self.web = color if color else 'black'
    File "/home/mark/.local/lib/python3.8/site-packages/colour.py", line 1006, in __setattr__
       fc(value)
    File "/home/mark/.local/lib/python3.8/site-packages/colour.py", line 1084, in set_web
       self.hex = web2hex(value)
    File "/home/mark/.local/lib/python3.8/site-packages/colour.py", line 664, in web2hex
       if web.startswith('#'):
    AttributeError: 'list' object has no attribute 'startswith'

startswith is a string method (not a list method), so I can understand why the error is happening.  Is there some modification you made to colour.py in order to accept lists?  I've tried to fix it myself, but I don't know manim's code well enough to know how to solve it, or your code for that matter (I can't even tell how it's even mixing between the two specified colors).  BTW, I had to guess a value for SKY_BLUE and did correctly modify constants.py

**Comment by nipunramk** [comment]
Hi Mark, good question, it may be that the version of manim that you are using is a little older than the one I was using at the time I made the FFT video. One workaround for older versions of manim is you can used the ```Mobject.set_color_by_gradient``` So ```edges[i].set_color_by_gradient([BLUE, SKY_BLUE])``` This should give the same behavior based on my local tests. 

P.S the ```SKY_BLUE``` is "#1919FF" if you want to get the exact same color. I have some plans to restructure the directory structure of this repo to make things easier for people like yourself to reproduce the scenes that I made, just haven't had the chance to do it, but should be able to in the few weeks so look out for that.

**Comment by mrmcsoftware** [comment]
Thanks for the response.  Unfortunately, I'm still getting the startswith error after changing it to set_color_by_gradient.  My version of manimlib should be relatively up-to-date since I only installed it a few weeks ago on a brand new Ubuntu 20.10 system using pip3 install.  But there is a chance there is something wrong with my installation since I have to put "from manim import *" after "from manimlib.imports import *" to solve a problem where only the first letter of a TextMobject is the desired color (and some lines would also generate out of range errors (especially the set_color lines in get_code)) and to solve an "ERROR                                   module_ops.py:6 There are no scenes inside that module." error caused by their deprecation of graph_scene.py

  My manimlib has deprecated a lot of things such as graph_scene.py moved to once_useful_constructs, and even worse "Passing Mobject methods to Scene.play is no longer supported. Use Mobject.animate instead." (many lines had to be changed).  TextMobject is also deprecated (use Tex or MathTex instead) but at least it still accepts TextMobject (with a warning).

I've tried various combinations of imports, but all combinations result in some error or undesired results, but the combination I mention in this reply gives the best results.

BTW, since you're thinking of restructuring, you might be interested in knowing the files earth, gps, and wc (possibly with py extensions) aren't provided.  Since they are "eye candy" so to speak, and not equations, etc. it's understandable that they're not in the repo.  Also, I got the MONOKAI colors from a website and put those in constants.py

In case you want to see my changes, here's the file: [fftmark.txt](https://github.com/nipunramk/Reducible/files/6133949/fftmark.txt)  I think all (or at least most) of my changes have MARK comments.

**Comment by nipunramk** [comment]
Hi Mark, so I tried running your version of FFT with my version of manim and the FFTGraph scene seems to work with no issues. So you are right that the issue is most likely something with the version/installation of manim that you have. I am not sure what could be causing the import issues you are encountering. The version of manim that I am using by the way is from commit 3c1a6bb8b10ec5e8c4760c91c9e9f148e81d3b57 (May 2020). Perhaps try pulling that version of manim into a separate repo, run this scene, and see if that resolves the issue. I will also be pushing my current version of manim (I have made a few changes on top of it, but none that should affect this scene) within the next few weeks as soon as I wrap up my current project, so hopefully that will help as well. I apologize I can't be of more help, let me know if you have any further questions.

**Comment by mrmcsoftware** [comment]
Using the version you mention (May 2020) solved every problem.  The most recent version won't work (because of the changes they made that I mentioned (graph_scene kinda gone and TextMobject renamed and maybe other changes).  The only reason I got as far as I did with what I previously had is the manim imports I added (which I have since found out is the Manim Community Edition) filled in the missing pieces (mostly).  But the May 2020 version has all the pieces that are missing in the newest version.  Thanks for your help!  I guess github etiquette is for me to close the issue now that it's solved.
